__tests__/apiMock/chunksToProperties.test.ts

```diff
@@ -0,0 +1,130 @@
+import { Chunk, Properties } from "../../lib/models";
+import { chunksToProperties } from "../../lib/utils";
+
+describe("chunksToProperties", () => {
+  it("converts an array of chunks into properties with type, description, and path correctly.", () => {
+    const chunks: Chunk[] = [
+      { path: ["id", "type"], data: "integer" },
+      { path: ["id", "description"], data: "Customer id" },
+    ];
+
+    const properties: Properties = chunksToProperties(chunks);
+
+    expect(properties).toEqual({
+      id: { type: "integer", description: "Customer id", path: ["id"] },
+    });
+  });
+
+  it("handles multiple paths correctly", () => {
+    const chunks: Chunk[] = [
+      { path: ["id", "type"], data: "integer" },
+      { path: ["id", "description"], data: "Customer id" },
+      { path: ["name", "type"], data: "string" },
+      { path: ["name", "description"], data: "Customer name" },
+    ];
+
+    const properties: Properties = chunksToProperties(chunks);
+    expect(properties).toEqual({
+      id: { type: "integer", description: "Customer id", path: ["id"] },
+      name: { type: "string", description: "Customer name", path: ["name"] },
+    });
+  });
+  it("multiple properties with the same name ", () => {
+    const chunks: Chunk[] = [
+      { path: ["id", "type"], data: "integer" },
+      { path: ["id", "description"], data: "Customer id" },
+      { path: ["name", "id", "type"], data: "string" },
+      { path: ["name", "id", "description"], data: "Customer name" },
+    ];
+
+    const properties: Properties = chunksToProperties(chunks);
+    expect(properties).toEqual({
+      id: { type: "integer", description: "Customer id", path: ["id"] },
+      "name.id": {
+        type: "string",
+        description: "Customer name",
+        path: ["name", "id"],
+      },
+    });
+  });
+
+  it("merges chunks with the same path", () => {
+    const chunks: Chunk[] = [
+      { path: ["id", "type"], data: "integer" },
+      { path: ["id", "description"], data: "Customer id" },
+      { path: ["id", "format"], data: "int64" },
+    ];
+
+    const properties: Properties = chunksToProperties(chunks);
+    expect(properties).toEqual({
+      id: { type: "integer", description: "Customer id", path: ["id"] },
```

Comment:
Confused by this test. Why does `format` disappear?

Reply:
we're not currently taking format into account when generating the response, only type

Reply:
Oh ok I thought `Properties` was a general way of reformulating json that would works on all jsons - so it's actually OpenAPI specific?

Reply:
yes queen, jsonDeconstruct / Reconstruct is general